src/daemon/daemon.cpp

```diff
@@ -135,7 +135,7 @@ bool t_daemon::run(bool interactive)
 
     mp_internals->p2p.run(); // blocks until p2p goes down
 
-    if (interactive)
+    if (rpc_commands)
```

Comment:
No, it is equivalent functionally, but this one will be clearer to the compiler.